report-viewer/src/components/ClusterGraph.vue

```diff
@@ -0,0 +1,177 @@
+  cluster: {
+    type: Object as PropType<ClusterListElement>,
+    required: true
+  }
+})
+
+const graphCanvas: Ref<HTMLCanvasElement | null> = ref(null)
+const loaded = ref(false)
+
+Chart.register(...registerables)
+Chart.register(ChartDataLabels)
+Chart.register(EdgeLine)
+Chart.register(GraphController)
+Chart.register(GraphChart)
+
+const keys = computed(() => Array.from(props.cluster.members.keys()))
+const labels = computed(() =>
+  Array.from(keys.value).map((m) =>
+    store().state.anonymous.has(m) ? 'Hidden' : store().submissionDisplayName(m) ?? m
+  )
+)
+const edges = computed(() => {
+  const edges: { source: number; target: number }[] = []
+  props.cluster.members.forEach((member, key) => {
+    member.forEach((match) => {
+      const firstIndex = keys.value.indexOf(key)
+      const secondIndex = keys.value.indexOf(match.matchedWith)
+      if (firstIndex < secondIndex) {
+        edges.push({ source: firstIndex, target: secondIndex })
+      }
+    })
+  })
+  return edges
+})
+
+function getSimilarityFromKeyIndex(firstIndex: number, secondIndex: number) {
+  const firstSubmission = props.cluster.members.get(keys.value[firstIndex])
+  if (!firstSubmission) {
+    return 0
+  }
+  const match = firstSubmission.find((m) => m.matchedWith == keys.value[secondIndex])
+  if (!match) {
+    return 0
+  }
+  return match.similarity
+}
+
+const graphData = computed(() => {
+  return {
+    labels: labels.value,
+    datasets: [
+      {
+        pointRadius: 10,
+        pointHoverRadius: 10,
+        pointBackgroundColor: graphColors.contentFill,
+        pointHoverBackgroundColor: graphColors.contentFill,
+        pointBorderColor: graphColors.ticksAndFont.value,
+        pointHoverBorderColor: graphColors.ticksAndFont.value,
+        data: Array.from(keys.value).map((_, index) => ({
+          x: Math.cos((2 * Math.PI * index) / keys.value.length) + 1,
+          y: Math.sin((2 * Math.PI * index) / keys.value.length) + 1
+        })),
+        edges: edges.value,
+        edgeLineBorderColor: (ctx: any) =>
+          graphColors.contentFillAlpha(getSimilarityFromKeyIndex(ctx.raw.source, ctx.raw.target)),
+        edgeLineBorderWidth: (ctx: any) =>
+          5 * getSimilarityFromKeyIndex(ctx.raw.source, ctx.raw.target) + 1
+      }
+    ]
+  }
+})
+
+const yPadding = 40
+const xPadding = computed(() => {
+  const avgCharacterLength = 9
+
+  const widths = labels.value.map((label) => label.length * avgCharacterLength)
+  const maxWidth = Math.max(...widths)
+  // Makes sure there is always space to display a name but the padding does not get too big
+  return Math.max(Math.min(200, maxWidth), 40)
+})
+
+const graphOptions = computed(() => {
+  return {
+    layout: {
+      padding: {
+        top: yPadding,
+        bottom: yPadding,
+        left: xPadding.value,
+        right: xPadding.value
+      }
+    },
+    animation: false as false,
+    plugins: {
+      legend: { display: false },
+      datalabels: {
```

Comment:
Is it possible to make the background color of the labels transparent? Sometimes, there is a little overlap of the label and the node of the graph which cuts of parts of the node's displayed circle

Reply:
Yes, this is what i meant. If there is no easy fix, let's ignore it for now